packages/application-shell/src/components/quick-access/quick-access.tsx

```diff
@@ -0,0 +1,508 @@
+import QuickAccessQuery from './quick-access.ctp.graphql';
+import createCommands from './create-commands';
+import { sanitize, translate } from './utils';
+import {
+  createProductTabsSubCommands,
+  createProductVariantSubCommands,
+} from './sub-commands';
+import messages from './messages';
+import { saveHistoryEntries, loadHistoryEntries } from './history-entries';
+import pimIndexerStates from './pim-indexer-states';
+import { ExecGraphQlQuery, Command, HistoryEntry } from './types';
+
+const searchProductIdsAction = (
+  searchText: string,
+  projectKey: string,
+  dataLocale: string
+) =>
+  sdkActions.post({
+    uri: `/${projectKey}/search/products`,
+    mcApiProxyTarget: MC_API_PROXY_TARGETS.PIM_SEARCH,
+    payload: {
+      query: {
+        fullText: {
+          field: 'name',
+          language: dataLocale,
+          value: searchText,
+        },
+      },
+      sort: [
+        {
+          field: 'name',
+          language: dataLocale,
+          order: 'desc',
+        },
+      ],
+      limit: 9,
+      offset: 0,
+    },
+  });
+const pimIndexerStatusAction = (
+  projectKey: string,
+  dataLocale: string // TODO this should be sdkActions.head()
+) =>
+  // and then we should check whether the response code is
+  // - 200 meaning the project is indexed
+  // - 404 meaning the project is not indexed
+  //
+  // But there is a problem in tne node-sdk client as it tries to
+  // .json()-parse the response to HEAD requests which results in an
+  // error, so we send a regular request for now and limit to no results
+  // instead to keep the payload minimal
+  sdkActions.post({
+    uri: `/${projectKey}/search/products`,
+    mcApiProxyTarget: MC_API_PROXY_TARGETS.PIM_SEARCH,
+    payload: {
+      query: {
+        fullText: {
+          field: 'name',
+          language: dataLocale,
+          value: 'availability-check',
+        },
+      },
+      limit: 0,
+      offset: 0,
+    },
+  });
+
+type FetchProductIdsResult = { hits?: { id: string }[] } | undefined;
+type Props = {
+  pimIndexerState: keyof typeof pimIndexerStates;
+  onPimIndexerStateChange: (
+    pimIndexerState: keyof typeof pimIndexerStates
+  ) => void;
+  onClose: () => void;
+  onChangeProjectDataLocale: (locale: string) => void;
+};
+
+const QuickAccess = (props: Props) => {
+  const [historyEntries, setHistoryEntries] = React.useState(
+    loadHistoryEntries()
+  );
+  const handleHistoryEntriesChange = React.useCallback<
+    (historyEntries: HistoryEntry[]) => void
+  >(historyEntries => {
+    setHistoryEntries(historyEntries);
+  }, []);
+  const apolloClient = useApolloClient();
+  const intl = useIntl();
+  const [
+    isPimSearchEnabled,
+    isCustomApplicationsEnabled,
+    isCanViewDashboardEnabled,
+  ] = useFeatureToggles({
+    pimSearch: true,
+    customApplications: true,
+    canViewDashboard: true,
+  });
+  const applicationContext = useApplicationContext();
+
+  const dispatchFetchProductIds = useAsyncDispatch<
```

Comment:
I kind of like our custom async dispatch function, makes it easy to use and get the correct types.